### PR TITLE
test(forms-e2e): feature file and tests for before you appeal page

### PR DESCRIPTION
**Pull request by davehlewis**
## Ticket Number

AS-1331

## Description of change
Feature file and step tests for Before Appeal guidance page

## Checklist

- [ ] Requires infrastructure changes
- [ ] If adding or remove environment variables (e.g. in `docker-compose.yaml`) then I have updated the appropriate Helm chart
- [ ] I have updated the documentation accordingly
- [X] My commit history in this PR is linear
- [X] New features have tests
- [ ] Breaking change (team conversation required)

## Important

Please do not merge from `master` (please only [rebase](https://github.com/foundry4/appeal-planning-decision/wiki/An-intro-to-Git-Rebase)). This keeps the history linear and easier to debug.

### CODE REVIEW

File: e2e-tests/cypress/integration/guidance-page-before-you-appeal.feature [hunk 1]
```diff
@@ -0,0 +1,21 @@
+Feature: Guidance Page - Before you Appeal
+  As a PO on the appeal service
+  I want prospective appellants to be aware of who the service is for
+  So that only those who need to appeal through the new appeal service do so
+
+  Scenario: Appellant navigates to next page
```
Comment (shilpabasu22) [inline e2e-tests/cypress/integration/guidance-page-before-you-appeal.feature:6]: It should be "
  Scenario: AC01 - Appellant navigates to next page"

File: e2e-tests/cypress/integration/guidance-page-before-you-appeal.feature [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    Given the appellant is on the before you appeal page
+    When the appellant navigates to the next page
+    Then information about when you can appeal is provided
+
+  Scenario Outline: Appellant selects <page> page from the Content list
```
Comment (shilpabasu22) [inline e2e-tests/cypress/integration/guidance-page-before-you-appeal.feature:11]: Same as above. It should be AC02

File: e2e-tests/cypress/integration/guidance-page-before-you-appeal.feature (hunk 1 above)
Comment (davehlewis) [reply e2e-tests/cypress/integration/guidance-page-before-you-appeal.feature:6]: Team 1 don't seem to mark their Scenarios like this so I was maintaining consistency. Should we treat this differently?

File: e2e-tests/cypress/integration/guidance-page-before-you-appeal.feature [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    Given the appellant is on the before you appeal page
+    When the appellant select a link from the content list: <page>
+    Then the appellant is navigated to that page: <url>
+
```
Comment (shilpabasu22) [inline e2e-tests/cypress/integration/guidance-page-before-you-appeal.feature:15]: Please check with @pippalou , that the current page shouldn't be a link in content list and we should check for it.

Comment (davehlewis) [reply e2e-tests/cypress/integration/guidance-page-before-you-appeal.feature:15]: This is the behaviour in the prototype

File: e2e-tests/cypress/integration/common/guidance-pages.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
+import { When, Then } from 'cypress-cucumber-preprocessor/steps';
+
+When('the appellant navigates to the next page', () => {
+  cy.guidancePageNavigation('next');
+});
+
+When('the appellant navigates to the previous page', () => {
```
Comment (shilpabasu22) [inline e2e-tests/cypress/integration/common/guidance-pages.js:7]: Why have we added this in current step definition

Comment (davehlewis) [reply e2e-tests/cypress/integration/common/guidance-pages.js:7]: We know it is needed generically for all so made sense to add it now

**Review by shilpabasu22 — CHANGES_REQUESTED** [review]
I have added inline comments.

File: e2e-tests/cypress/integration/guidance-page-before-you-appeal.feature (hunk 1 above)
Comment (shilpabasu22) [reply e2e-tests/cypress/integration/guidance-page-before-you-appeal.feature:6]: as far as I know, they had started following this suite later on in development. We have been maintaining this from beginning.

File: e2e-tests/cypress/integration/guidance-page-before-you-appeal.feature (hunk 3 above)
Comment (shilpabasu22) [reply e2e-tests/cypress/integration/guidance-page-before-you-appeal.feature:15]: I agree, I was in a joint review with @pippalou and thus I have asked you to check with her.

File: e2e-tests/cypress/integration/common/guidance-pages.js (hunk 4 above)
Comment (shilpabasu22) [reply e2e-tests/cypress/integration/common/guidance-pages.js:7]: But we shouldn't write any extra code even though we know that it will be used later.

**Review by shilpabasu22 — APPROVED** [review]

**Pull request merged**